core/common/src/main/java/alluxio/uri/SingleMasterAuthority.java

```diff
@@ -66,11 +63,11 @@ public boolean equals(Object o) {
 
   @Override
   public int hashCode() {
-    return Objects.hashCode(mAuthority);
+    return Objects.hashCode(toString());
   }
 
   @Override
   public String toString() {
-    return mAuthority;
+    return String.format("%s:%d", mHost, mPort);
```

Comment:
`compareTo` and `equals` could be much cheaper if we directly compare the host/port fields instead of calling `toString()` and constructing a new string

Reply:
So then for compareTo how exactly should we be determining the return value if we are comparing the two separately? I know how the string comparison works, but I don't exactly how I would be computing the return value from the separate host and port.

My first thought thought for this would be


```java
int comp = mHost.compareTo(otherHost);
if (comp == 0) {
  // Compare ports
  if (mPort < otherPort) {
  return -1;
} else if (mPort > otherPort) {
    return 1;
  } else {
    return 0
  }
} else {
  return comp;
}
```

Reply:
This approach looks nice: https://stackoverflow.com/a/25501226/401884

Reply:
Ah, thought it was `Comparable<SingleMasterAuthority>`. Agree we need to stick with `toString()`